flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/rest/SqlGatewayRestEndpointFactory.java

```diff
@@ -94,6 +92,10 @@ public Set<ConfigOption<?>> requiredOptions() {
 
     @Override
     public Set<ConfigOption<?>> optionalOptions() {
-        return Collections.emptySet();
+        Set<ConfigOption<?>> options = new HashSet<>();
+        options.add(BIND_ADDRESS);
+        options.add(PORT);
+        options.add(BIND_PORT);
+        return options;
```

Comment:
Adding code here is for the options check. 
These options are defined to match the logic of RestServerEndpoint reading configuration.

By default, the user only need to select a local address to set ADDRESS, then the server will bind the address to all the local IPV4 address (0.0.0.0) and bind the port to BIND_PORT(fallback to the default value of PORT). In the log, the server will suggest client to connect to ADDRESS with the default value of PORT.

1. If user specifies BIND_ADDRESS, then the server will bind to BIND_ADDRESS and suggest the user that the current client should connect to this BIND_ADDRESS in the log rather than ADDRESS.

2. If user specifies PORT, then the server will bind the port to BIND_PORT(fallback to the value of PORT).

3. If user specifies BIND_PORT, then the server will ignore PORT and  directly bind the port to the value of BIND_PORT.

Reply:
Thanks for your illustration. +1 for the change. Could you describe the behaviour in the config options.